scipy/special/xsf/specfun/specfun.h

```diff
@@ -5112,17 +5114,19 @@ void rmn2sp(int m, int n, T c, T x, T cv, int kd, T *df, T *r2f, T *r2d) {
     }
     su2 = 0.0;
     ki = (2 * m + 1 + ip) / 2;
+    ki = std::max(1, ki);
+    assert((ki-1) >= 0);
```

Comment:
I think this is a bug in f77.

With `pro_rad2(0, 0, 3, 1.02)`, we have `ki = 0, nm3 = 28`,
then `dn[k - 1]` may result in index out-of-bounds.

Using `max` to limit `ki` to a minimum of 1 is the easiest fix. The function also gives the correct result after the fix.
But I'm not sure if there is a better way, or if this has any effect on the precision.

 
https://github.com/scipy/scipy/blob/9d08eb58205fb669ff72c60080b18be61b14d8e1/scipy/special/specfun/specfun.f#L668-L674